src/protobuf/reader.ts

```diff
@@ -1,41 +1,41 @@
-import { Buffer } from "buffer";
-
 export class ProtoBufReader {
-  buf: Buffer;
+  buf: Uint8Array;
   c: number;
   s: number = 0;
 
   static splitHeader(n: bigint): [bigint, number] {
     return [n >> 3n, Number(n & 0x7n)];
   }
 
-  static parseVariant(buf: Buffer): bigint {
+  static parseVariant(buf: Uint8Array): bigint {
     return buf.reduce(
       (r, b, i) => r | ((BigInt(b) & 0x7fn) << (BigInt(i) * 7n)),
       0n
     );
   }
 
-  constructor(buf: Buffer) {
+  constructor(buf: Uint8Array) {
     this.buf = buf;
     this.c = 0;
   }
 
-  eat(bytes: number): Buffer | null {
+  eat(bytes: number): Uint8Array | null {
     if (this.isEnded()) return null;
     return this.buf.slice(this.c, (this.c += bytes));
   }
 
   eatUInt32(): number | null {
     if (this.isEnded()) return null;
-    const n = this.buf.readUInt32LE(this.c);
+    const n = new DataView(this.buf).getUint32(this.c, true);
```

Comment:
This is super inefficient, but I'd like to keep it this way for simplicity sake (it's a 1:1 translation of what it was doing with `Buffer` before). Same in `eatUInt64`